src/elements/content-sidebar/__tests__/SidebarNavSign.test.tsx

```diff
@@ -1,80 +1,58 @@
-        render(
-            <FeatureProvider features={features}>
-                <SidebarNavSign {...props} />
-            </FeatureProvider>,
-        );
+        render(<SidebarNavSign {...props} />, {
+            wrapperProps: { features },
+        });
 
-    test.each([true, false])('should render sign button', isRemoveInterstitialEnabled => {
-        const features = {
-            boxSign: {
-                isSignRemoveInterstitialEnabled: isRemoveInterstitialEnabled,
-            },
-        };
+    test('should render sign button', async () => {
+        renderComponent();
 
-        const wrapper = renderComponent({}, features);
-        expect(wrapper.getByTestId('sign-button')).toBeVisible();
+        expect(screen.getByRole('button', { name: 'Request Signature' })).toBeVisible();
     });
 
-    test('should call correct handler when sign button is clicked', () => {
-        const features = {
-            boxSign: {
-                isSignRemoveInterstitialEnabled: false,
-                onClick: onClickRequestSignature,
-            },
-        };
-        const { getByTestId } = renderComponent({}, features);
-
-        fireEvent.click(getByTestId('sign-button'));
+    test('should open dropdown with 2 menu items when sign button is clicked', async () => {
+        renderComponent();
 
-        expect(onClickRequestSignature).toBeCalled();
-    });
+        await userEvent.click(screen.getByRole('button', { name: 'Request Signature' }));
 
-    test('should open dropdown with 2 menu items when sign button is clicked', () => {
-        const features = {
-            boxSign: {
-                isSignRemoveInterstitialEnabled: true,
-            },
-        };
-        const { getByTestId } = renderComponent({}, features);
-        fireEvent.click(getByTestId('sign-button'));
-        expect(getByTestId('sign-request-signature-button')).toBeVisible();
-        expect(getByTestId('sign-sign-myself-button')).toBeVisible();
+        expect(screen.getByText('Request Signature')).toBeVisible();
+        expect(screen.getByText('Sign Myself')).toBeVisible();
     });
 
-    test('should call correct handler when request signature option is clicked', () => {
+    test('should call correct handler when request signature option is clicked', async () => {
         const features = {
             boxSign: {
-                isSignRemoveInterstitialEnabled: true,
                 onClick: onClickRequestSignature,
             },
         };
-        const { getByTestId } = renderComponent({}, features);
-        fireEvent.click(getByTestId('sign-button'));
-        fireEvent.click(getByTestId('sign-request-signature-button'));
-        expect(onClickRequestSignature).toBeCalled();
+        renderComponent({}, features);
+
+        await userEvent.click(screen.getByRole('button', { name: 'Request Signature' }));
+        await userEvent.click(screen.getByText('Request Signature'));
```

Comment:
how come we are clicking on the button and the text but only asserting if one callback function was called?

Reply:
first we are clicking the button that triggers the modal, and in that modal we have the options "Request a signature" and "Sign Myself". What happens is that the `aria-label` of the button coincides with the dropdown item text

Reply:
are we able to `await userEvent.click(screen.getByRole('menu-item', { name: 'Request Signature' }));` instead to at least clearly description the act

Reply:
yes, it’s possible to use `getByRole` for the menu items. I’ve included that update in the latest changes I just pushed